CMakeLists.txt

```diff
@@ -19,10 +22,6 @@ if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-if(NOT CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel|GNU)$")
-  message(WARNING "Compiler not officially supported: ${CMAKE_Fortran_COMPILER_ID}")
-endif()
-
```

Comment:
The code is not pure Fortran. The different versions still use compile flags for kinds.

Reply:
How does this warning help that?

Reply:
It doesn't, so maybe a `message(FATAL_ERROR "Compiler not supported")` would be better? I don't think it's helpful to incorrectly build the library. Your `_4` version of the library may be correct but it'll loop over non-existent variables looking for `-r8/-fdefault-real-8` type compiler flags.

Reply:
OK, I have readded the warning.

We should not make this a fatal error, because, with correct flags, the code can work for other compilers.